src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs

```diff
@@ -134,7 +152,8 @@ public async Task InitializeDataStoreAsync(CosmosClient client, CosmosDataStoreC
                                 cancellationToken: cancellationToken));
                 }
 
-                foreach (var collectionInitializer in collectionInitializers)
+                // Manged identity does not have permission to read/update stored procedures or change collection settings
+                foreach (ICollectionInitializer collectionInitializer in collectionInitializers)
```

Comment:
Should this section have an if statement checking for using managed identity?

Reply:
There are some initializers that populate data